common/src/gadget/substrate/mod.rs

```diff
@@ -0,0 +1,299 @@
+use std::ops::Deref;
+use std::sync::Arc;
+use async_trait::async_trait;
+use sp_core::{ecdsa, keccak_256, sr25519};
+use tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::{jobs, roles};
+use gadget_core::gadget::substrate::FinalityNotification;
+use gadget_core::job_manager::{PollMethod, WorkManagerInterface};
+use tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_testnet_runtime::{MaxAdditionalParamsLen, MaxParticipants, MaxSubmissionLen};
+use gadget_core::gadget::manager::AbstractGadget;
+use crate::config::GadgetProtocol;
+use crate::gadget::{EventHandler, GeneralModule, MetricizedJob};
+use crate::prelude::{ClientWithApi, GadgetProtocolMessage, WorkManager};
+use crate::protocol::AsyncProtocol;
+use crate::tangle_runtime::AccountId32;
+
+/// Endow the general module with event-handler specific code tailored towards interacting with substrate
+pub struct SubstrateGadget<C, N, M> {
```

Comment:
Let's call this the Tangle gadget, immediately in the event handler you're using Tangle runtime to call into our job framework. It is much more specific than just Substrate.. it's Tangle. If we are to build a connection to Eigenlayer in the future for parsing AVS jobs it would be EigenlayerGadget not EthereumGadget.